### DISCUSSION TITLE
Error when updating relational field

### DISCUSSION THREAD

**Discussion by terion-name** [directus/directus#13584] [Archive]
### Describe the Bug

In continuation of #13369 (relational structure is there): because of relation duplication bug, decided to make models immutable. Added a condition to relational field and got:
```
[INTERNAL_SERVER_ERROR] alter table `order_slots` modify `item` int - UNKNOWN_CODE_PLEASE_REPORT: Referencing column 'item' and referenced column 'id' in foreign key constraint 'order_slots_item_foreign' are incompatible.
```

### To Reproduce

See screencast:

https://user-images.githubusercontent.com/1060205/169021451-0ad122df-9488-40ac-80da-2240da9fb4d2.mp4

### Errors Shown

[INTERNAL_SERVER_ERROR] alter table `order_slots` modify `item` int - UNKNOWN_CODE_PLEASE_REPORT: Referencing column 'item' and referenced column 'id' in foreign key constraint 'order_slots_item_foreign' are incompatible.

### What version of Directus are you using?

9.10.0

### What version of Node.js are you using?

v16.14.0

### What database are you using?

mysql

### What browser are you using?

brave

### How are you deploying Directus?

locally

**Comment by rijkvanzanten** [comment]
Could you confirm that the collation is the same across all tables that are used in this relationship? It reminds me of https://github.com/directus/directus/issues/9897 which was inherently caused by the referencing table being of a different collation in MySQL.

**Reply by terion-name** [reply]
they are
